### PR TITLE
Internals Refactor: Seclude service configuration parsing out of internals

### PR DESCRIPTION
**Pull request by medikoo**
Addresses 2.1 from #8364

In a side PR  (https://github.com/serverless/test/pull/71) it's ensured that `runServerless` adapts to new way of serverless config path resolution and passes  `configuration` to `Serverless` constructor (still it's not mandatory for successful test runs, so both PR's can be handled independently)

Additionally (all independent updates where submitted with individual commits):
- Revert from introducing legacy _sync_ utils to catch up on missing options in constructor. As in this case it was not possible to introduce _sync_ counterpart, and I've realized it's good enough to fill in missing properties at begin of `serverless.init` (so original utils can be used)
- Introduce standalone `isHelpRequest` and use internally instead of CLI class method, which is removed
- Apply async/await refactor to touched parts
- Fix accidentally introduced regression with #8770 with which autocomplete didn't resolve expected hash for service configuration

### CODE REVIEW

File: scripts/serverless.js [hunk 1]
```diff
@@ -37,8 +37,25 @@ const processSpanPromise = (async () => {
...
+    const readConfiguration = require('../lib/configuration/read');
 
-    serverless = new Serverless({ configurationPath: await resolveConfigurationPath() });
+    const configurationPath = await resolveConfigurationPath();
+    const configuration = configurationPath
+      ? await (async () => {
+          try {
+            return await readConfiguration(configurationPath);
+          } catch (error) {
+            if (isHelpRequest()) return null;
```
Comment (pgrzesik) [inline scripts/serverless.js:49]: Is there a reason to check for `isHelpRequest` only during errored out `readConfiguration` call? Maybe there's not need for resolving path and loading configuration in such case and we could introduce that check before `resolveConfigurationPath` ?

**Review by pgrzesik — APPROVED** [review]
Looks awesome, great job :+1: I've added one comment out of curiosity, but it's a non-blocker

File: scripts/serverless.js (hunk 1 above)
Comment (medikoo) [reply scripts/serverless.js:49]: Good question. Reason for that is that if there are plugins in a _service_ then they influence _help_ output (commands they introduce are listed in help)

So we also want to parse config if it's a _help_ request, but failure of parse should not prevent display of help

Comment (pgrzesik) [reply scripts/serverless.js:49]: Understood, thanks for the explanation :bow:

Comment (medikoo) [reply scripts/serverless.js:52]: I've added comment explaining that.

**Review by pgrzesik — APPROVED** [review]

**Pull request merged**